server/midddleware/auth.go

```diff
@@ -115,32 +107,20 @@ func AuthFunction(ctx context.Context, jwtValidator *validator.Validator, config
 		return ctx, err
 	}
 
-	validToken, err := jwtValidator.ValidateToken(ctx, tkn)
+	validatedToken, err := jwtValidator.ValidateToken(ctx, tkn)
 	if err != nil {
 		return ctx, api.Errorf(api.Code_UNAUTHENTICATED, err.Error())
 	}
-
-	log.Debug().Msg("Valid token received")
-	return context.WithValue(ctx, TokenCtxkey{}, validToken), nil
-}
-
-func getOrganizationName(ctx context.Context) (string, error) {
-	host := api.GetHeader(ctx, ":authority")
-	if host == "" {
-		host = api.GetHeader(ctx, "host")
-	}
-	// <project>-<org-name>.<env>.tigrisdata.cloud
-	parts := strings.Split(host, ".")
-	if len(parts) < 3 {
-		return "", api.Errorf(api.Code_FAILED_PRECONDITION, "hostname is not as per expected scheme")
-	}
-	subParts := strings.Split(parts[0], "-")
-	if len(subParts) > 2 {
-		return "", api.Errorf(api.Code_FAILED_PRECONDITION, "hostname is not as per expected scheme")
-	}
-	organizationName := subParts[0]
-	if len(subParts) > 1 {
-		organizationName = subParts[1]
+	if validatedClaims, ok := validatedToken.(*validator.ValidatedClaims); ok {
+		if customClaims, ok := validatedClaims.CustomClaims.(*CustomClaim); ok {
+			log.Debug().Msg("Valid token received")
+			token := v1.AccessToken{
+				Namespace: customClaims.Namespace.Code,
+				Sub:       validatedClaims.RegisteredClaims.Subject,
+			}
+			result := v1.SetAccessToken(ctx, token)
+			return result, nil
+		}
 	}
-	return organizationName, nil
+	return ctx, api.Errorf(api.Code_UNAUTHENTICATED, err.Error())
```

Comment:
there is no error at this point i.e. err obj is nil. Don't we need to send some custom message here?

Reply:
at this point, the claims in token are not validataed. so user is still unauthorized. Good point, let me update message.

Reply:
that is fine but my point was there is no JWT error in the method at this point because there is a check-in line 110 and based on that you're already returning. At this point, the error needed to be a custom msg.

Reply:
yes this line should never be invoked. updated message and comment.